coalib/output/ConsoleInteraction.py

```diff
@@ -922,27 +933,49 @@ def print_bears(bears,
     :param show_params:      True if the parameters and their description
                              should be shown.
     :param console_printer:  Object to print messages on the console.
+    :param args:             Args passed to coala command.
     """
     if not bears:
         console_printer.print('No bears to show. Did you forget to install '
                               'the `coala-bears` package? Try `pip3 install '
                               'coala-bears`.')
         return
 
-    for bear, sections in sorted(bears.items(),
-                                 key=lambda bear_tuple:
-                                 bear_tuple[0].name.lower()):
-        show_bear(bear,
-                  show_description,
-                  show_params,
-                  console_printer)
+    results = [bear for bear, _ in sorted(bears.items(),
+                                          key=lambda bear_tuple:
+                                          bear_tuple[0].name.lower())]
+    if args and args.json:
+        from coalib.output.JSONEncoder import create_json_encoder
+        JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+        json_output = {'bears': results}
+        import json
+        json_formatted_output = json.dumps(json_output,
+                                           cls=JSONEncoder,
+                                           sort_keys=True,
+                                           indent=2,
+                                           separators=(',', ': '))
+        if args.output:
+            filename = str(args.output[0])
```

Comment:
`args.output[0]`? Why not `args.output`?

Reply:
`args.output` is list of file names given as args. It is used in similar manner in other places (https://github.com/coala/coala/pull/4726/commits/a581d099b1d8b8dab3a910f3ef2e2a974e67338f#diff-a297897b536e74044ca36e47697b169dL83) as well.

Reply:
@Makman2

Reply:
Shouldn't we then iterate over each file and write to each of them??

Reply:
True. But do we really need an option like `output` ?
If user is using cli, there's always an option to use Linux i/o redirectives. It's actually easy to remember and shorter to write.

Reply:
though we should still support it for now (with multiple files) and discussing later the usefulness of `--output`

Reply:
Ah and you should adapt tests to use multiple output files to test that properly ;)

Reply:
Ah! Only one argument is supported after `--output` using the parameter `nargs=1` while building parser. But it is stored in a list by default irrespective of number of args.
Do we still want to implement this for multiple files ?

Reply:
Ah because of `nargs`... I think if someone omits `nargs=1` in the parameter definition altogether, it requires by default a single value and doesn't use an array to store it. However, this is not job of this PR to change that, so keep it ;)

Reply:
Maybe add a comment then? Not sure...